redbot/cogs/mod/mutes.py

```diff
@@ -251,27 +253,36 @@ class MuteMixin(MixinMeta):
         author = ctx.message.author
         guild = ctx.guild
         audit_reason = get_audit_reason(author, reason)
+        guild_success = True
 
         mute_success = []
         for channel in guild.channels:
             success, issue = await self.mute_user(guild, channel, author, user, audit_reason)
             mute_success.append((success, issue))
             await asyncio.sleep(0.1)
-        try:
-            await modlog.create_case(
-                self.bot,
-                guild,
-                ctx.message.created_at,
-                "smute",
-                user,
-                author,
-                reason,
-                until=None,
-                channel=None,
-            )
-        except RuntimeError as e:
-            await ctx.send(e)
-        await ctx.send(_("User has been muted in this server."))
+
+            if success is False:  # Check the latest entry
+                guild_success = False
+                break
```

Comment:
This will break if the bot is explicitly denied permissions to handle overwrites inside any channels when trying to mute a user rendering the command useless in a majority of properly setup permission servers. It would be better to handle this after attempting all channels and checking teh `mute_success` list for displaying where things went wrong. Or you may change the return from `mute_user` to be more than just a boolean return statement and then handle things based on severity of the error that way.

Reply:
Just changing the return value to `None` should suffice, right?

Reply:
Yes, it would.

Reply:
There is 1 place where the return value is None but not where we want to explicitly stop the funciton from continuing.

Reply:
But the other things to consider is that a server has bot in an admin channel at the top of the server channel list and the bot is explictly denied permission to manage_roles in that channel. In the current way this is setup the entire mute funciton will fail silently without providing the overwrites to subsequent channels which is expected.

Reply:
You're right in that regard and I suppose I'll redo some stuff later.